### PR TITLE
doc: document user defined functions (UDFs)

### PR DESCRIPTION
**Pull request by annastuchlik**
This PR is V2 of the[ PR created by @psarna.](https://github.com/scylladb/scylladb/pull/11560).
I have:
- copied the content.
- applied the suggestions left by @nyh.
- made minor improvements, such as replacing "Scylla" with "ScyllaDB", fixing punctuation, and fixing the RST syntax.

Fixes https://github.com/scylladb/scylladb/issues/11378

### CODE REVIEW

**Comment by annastuchlik** [comment]
@nyh @tzach Could you do the final review? You've already reviewed it in https://github.com/scylladb/scylladb/pull/11560, but I've made the requested changes.

**Comment by tzach** [comment]
Looks good to me, but I have not run the functions.

I have a problem with the Apache Cassandra copyright at the footer.
While the original file was copied from Apache Cassandra and needed copyright, the new content, like this one, does not.
One solution is to move it to a new page without copyright and cross-link.

File: docs/cql/functions.rst [hunk 1]
```diff
@@ -228,6 +229,65 @@ A number of functions are provided to “convert” the native types into binary
...
+keywords are used, the function will only be created only if another function with the same
+signature does not exist. ``OR REPLACE`` and ``IF NOT EXISTS`` cannot be used together.
+
+Behavior for null input values must be defined for each function:
+
+* ``RETURNS NULL ON NULL INPUT`` declares that the function will always return null (without being executed) if any of the input arguments is null.
+* ``CALLED ON NULL INPUT`` declares that the function will always be executed.
+
+Function Signature
+``````````````````
```
[the remark this answers is not in the document]
Comment (annastuchlik) [reply docs/cql/functions.rst:270]: Great idea! Shouldn't we add that tag to the entire User-defined functions section? According to the note added by Piotr, all user-defined functions are experimental.

[the remark this answers is not in the document]
Comment (tzach) [reply docs/cql/functions.rst:270]: Yes, all UDF are experimental

**Comment by annastuchlik** [comment]
@tzach Most of the content provided by Piotr was copied from the [Cassandra documentation.](https://cassandra.apache.org/doc/latest/cassandra/cql/functions.html)
One difference is that Cassandra supports Java and JavaScriprit and includes examples for Java, and we support Lua and WebAssembly - and the examples are based on Lua.
I know this makes a difference, but moving the new content to a separate page may make it harder to find information. For example, "User-defined aggregates (UDAs)" are logically a subsection of "Aggregate functions".

I'd leave it as is unless you insist on creating a separate page or there are some legal aspects involved. Please let me know.

**Comment by tzach** [comment]
OK, lets leave it with the copyright as is

**Comment by annastuchlik** [comment]
@nyh Could you review it so that we can merge it? You've already reviewed V1 added by Piotr. This V2 only added some minor changes you suggested.

**Comment by annastuchlik** [comment]
@scylladb/scylla-maint Please backport to 5.1

**Comment by annastuchlik** [comment]
@scylladb/scylla-maint Please backport to 5.1

**Comment by denesb** [comment]
> @scylladb/scylla-maint Please backport to 5.1

Done.

**Pull request merged**